src/python/grpcio/tests/unit/framework/common/test_control.py

```diff
@@ -72,19 +73,31 @@ def control(self):
       if self._fail:
         raise Defect()
 
-      while self._paused:
+      while self._pause:
+        self._paused = True
+        self._condition.notify_all()
         self._condition.wait()
+      self._paused = False
 
   @contextlib.contextmanager
   def pause(self):
     """Pauses code under control while controlling code is in context."""
     with self._condition:
-      self._paused = True
+      self._pause = True
     yield
     with self._condition:
```

Comment:
I'm not sure what you're asking here - read up on what `@contextlib.contextmanager` does to this function?

(One of the reasons I don't like decorators in Python is that they often change code in radical ways.)

I think this object makes sense with callable public methods and non-callable private fields and I'm not at all seeing what a private method would do?